src/message-io.js

```diff
@@ -85,18 +85,29 @@ module.exports = class MessageIO extends EventEmitter {
     return this._packetSize;
   }
 
-  startTls(credentialsDetails) {
+  startTls(credentialsDetails, trustServerCertificate) {
     const credentials = tls.createSecureContext ? tls.createSecureContext(credentialsDetails) : crypto.createCredentials(credentialsDetails);
 
     this.securePair = tls.createSecurePair(credentials);
     this.tlsNegotiationComplete = false;
 
-    this.securePair.on('secure', () => {
-      const cipher = this.securePair.cleartext.getCipher();
-      this.debug.log('TLS negotiated (' + cipher.name + ', ' + cipher.version + ')');
-      this.emit('secure', this.securePair.cleartext);
-      this.encryptAllFutureTraffic();
-    });
+    this.securePair.on('secure', (_this) => {
```

Comment:
It will not, due to the fact that the arrow function `( (_this)=>{ return function() { } } )( this )` ends up returning `function(){}` with `_this` accessible, where `_this` is the outer `this` due to being passed in as argument.

That more traditional function then has the `this` scope from the `secure` event ( arrow functions do not have this, they do not have their own this scope )

To paraphrase from mozilla, `An arrow function expression ... does not bind it's own this, arguments, super, or new.target` - however it is the functions own `this` that I need bound.